runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -350,8 +351,20 @@ private Options getPersistentMapOptions() {
     Table<K, V, M> getTable(String namespace, String tableName) {
         String fullyQualifiedTableName = getFullyQualifiedTableName(namespace, tableName);
         if (!tableMap.containsKey(fullyQualifiedTableName)) {
-            throw new NoSuchElementException(
-                    String.format("No such table found: namespace: %s, tableName: %s", namespace, tableName));
+            // Table has not been opened, but let's first find out if this table even exists
+            // in the corfu cluster
+            UUID tableStreamId = CorfuRuntime.getStreamID(fullyQualifiedTableName);
+            long tableStreamTail = runtime.getSequencerView().query(tableStreamId);
+            if (tableStreamTail != Address.NON_EXIST) {
+                // If table does exist then the caller must use the long form of the openTable()
+                // since there are too few arguments to open a table not seen by this runtime.
+                throw new IllegalArgumentException("Insufficient parameters to re-open existing table "
```

Comment:
should this message be a bit more informative? as it states "unable to 'reopen' existing table" it does not provide a notion that this table has never been opened, instead it gives the idea that it has been opened but as of now we can't reopen.

Reply:
Right now the caller is just polling on the Exception returned by this api - IllegalArgumentException vs NoSuchElementException. But sure should I instead say please provide Key, Value and Metadata schemas to re-open the existing table?